python/cudf/cudf/core/index.py

```diff
@@ -521,6 +545,132 @@ def get_loc(self, key, method=None, tolerance=None):
             raise KeyError(key)
         return np.clip(round_method(idx), 0, idx_int_upper_bound, dtype=int)
 
+    def _union(self, other, sort=None):
+        if isinstance(other, RangeIndex):
+            start_s, step_s = self.start, self.step
+            end_s = self.start + self.step * (len(self) - 1)
+            start_o, step_o = other.start, other.step
+            end_o = other.start + other.step * (len(other) - 1)
+            if self.step < 0:
+                start_s, step_s, end_s = end_s, -step_s, start_s
+            if other.step < 0:
+                start_o, step_o, end_o = end_o, -step_o, start_o
+            if len(self) == 1 and len(other) == 1:
+                step_s = step_o = abs(self.start - other.start)
+            elif len(self) == 1:
+                step_s = step_o
+            elif len(other) == 1:
+                step_o = step_s
+            start_r = min(start_s, start_o)
+            end_r = max(end_s, end_o)
+            result = None
+            if step_o == step_s:
+                if (
+                    (start_s - start_o) % step_s == 0
```

Comment:
I think you can early exit (skip the entire `if step_o == step_s`) by moving this check above it and using `min(step_o, step_s)`. I'm envisioning something like
```
min_step = min(step_o, step_s)
steps_commensurate = `(start_s - start_o) % min_step`
if steps_commensurate:
    if step_o == step_s:
...
```
That would help improve the readability of the conditionals, I think.